### PR TITLE
[Yaml] Add a test adding a new fabric from an existing fabric

### PR DESCRIPTION
**Pull request by vivien-apple**
#### Problem

This PR adds some commands to chip-tool to get the root certificate for a given identity and allow to generate a NOC chain from a previous CSRRequest.
It also add a YAML file that exercise those commands using the chip-tool yaml python runner.

### CODE REVIEW

File: scripts/tests/chiptest/__init__.py [hunk 1]
```diff
@@ -133,7 +133,9 @@ def _GetInDevelopmentTests() -> Set[str]:
        Currently this is empty and returns an empty set, but this is kept around in case
        there are tests that are a work in progress.
     """
-    return set()
+    return {
+        "Test_AddNewFabricFromExistingFabric.yaml",     # chip-repl does not supports GetCommissionerRootCertificate and IssueNocChain command
```
Comment (bzbarsky-apple) [inline scripts/tests/chiptest/__init__.py:137]:
```suggestion
        "Test_AddNewFabricFromExistingFabric.yaml",     # chip-repl does not support GetCommissionerRootCertificate and IssueNocChain command
```

File: src/app/tests/suites/Test_AddNewFabricFromExistingFabric.yaml [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+      arguments:
+          values:
+              - name: "NOCValue"
+                value: noc
+              - name: "ICACValue"
+                value: icac
+              - name: "IPKValue"
+                value: ipk
+              - name: "CaseAdminSubject"
+                value: 0x000000000001B669
```
[the remark this answers is not in the document]
Comment (vivien-apple) [reply src/app/tests/suites/Test_AddNewFabricFromExistingFabric.yaml]: I have used `GetCommissionerNodeId`.

File: scripts/tests/chiptest/__init__.py (hunk 1 above)
Comment (vivien-apple) [reply scripts/tests/chiptest/__init__.py]: Done.

File: examples/chip-tool/commands/pairing/ToChipCert.h [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+ *
+ */
+
+#pragma once
+
+#include <lib/support/Span.h>
+#include <string>
+
+CHIP_ERROR ToBase64(const chip::ByteSpan & input, std::string & output);
+CHIP_ERROR ToChipCert(const chip::ByteSpan & input, std::string & output);
```
[the remark this answers is not in the document]
Comment (vivien-apple) [reply examples/chip-tool/commands/pairing/ToChipCert.h]: I have updated the argument names. Let me know if that is enough.

File: examples/chip-tool/commands/pairing/IssueNOCChainCommand.h [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+#include "ToChipCert.h"
+
+class IssueNOCChainCommand : public CHIPCommand
+{
+public:
+    IssueNOCChainCommand(CredentialIssuerCommands * credIssuerCommands) :
+        CHIPCommand("issue-noc-chain", credIssuerCommands), mDeviceNOCChainCallback(OnDeviceNOCChainGeneration, this)
+    {
+        AddArgument("elements", &mNOCSRElements);
```
[the remark this answers is not in the document]
Comment (vivien-apple) [reply examples/chip-tool/commands/pairing/IssueNOCChainCommand.h]: Yes. I have added a command help.

File: examples/chip-tool/BUILD.gn [hunk 5]
```diff
@@ -48,6 +48,12 @@ static_library("chip-tool-utils") {
   sources = [
     "${chip_root}/src/app/tests/suites/include/ConstraintsChecker.h",
     "${chip_root}/src/app/tests/suites/include/ValueChecker.h",
+
+    # TODO - enable CommissionedListCommand once DNS Cache is implemented
+    #    "commands/pairing/CommissionedListCommand.cpp",
+    #    "commands/pairing/CommissionedListCommand.h",
```
[the remark this answers is not in the document]
Comment (vivien-apple) [reply examples/chip-tool/BUILD.gn]: I have opened https://github.com/project-chip/connectedhomeip/pull/25998 for that part. This is somehow unrelated.

**Review by bzbarsky-apple — APPROVED** [review]

**Review by andy31415 — APPROVED** [review]

**Pull request merged**